app/src/views/private/components/files-navigation-folder.vue

```diff
@@ -154,6 +156,28 @@ function useDeleteFolder() {
 		}
 	}
 }
+
+async function downloadFolder() {
+	const response = await fetch(getFolderUrl(props.folder.id), {
+		method: 'POST',
+		headers: {
+			'Content-Type': 'application/json',
+		},
+	});
+
+	if (!response.ok) {
+		unexpectedError({ response: { data: await response.json() } });
+	}
+
+	const blob = await response.blob();
+	const filename = response.headers.get('Content-Disposition')?.match(/filename="(.*?)"/)?.[1];
+
+	const url = window.URL.createObjectURL(blob);
+	const a = document.createElement('a');
+	a.href = url;
+	a.download = filename ?? `folder-unknown-${getDateTimeFormatted()}.zip`;
+	a.click();
```

Comment:
Please refer to this [comment ](https://github.com/directus/directus/pull/26006#discussion_r2601782780)